xhydro/testing/helpers.py

```diff
@@ -0,0 +1,205 @@
+"""Helper functions for testing data management."""
+
+import importlib.resources as ilr
+import logging
+import os
+from pathlib import Path
+from typing import Optional, Union
+from urllib.parse import urljoin
+
+import pooch
+
+from xhydro import __version__ as __xhydro_version__
+
+__all__ = [
+    "DATA_DIR",
+    "DATA_URL",
+    "DEVEREAUX",
+    "generate_registry",
+    "load_registry",
+    "populate_testing_data",
+]
+
+_default_cache_dir = pooch.os_cache("xhydro-testdata")
+
+DATA_DIR = os.getenv("XHYDRO_DATA_DIR", _default_cache_dir)
+"""Sets the directory to store the testing datasets.
+
+If not set, the default location will be used (based on ``platformdirs``, see :func:`pooch.os_cache`).
+
+Notes
+-----
+When running tests locally, this can be set for both `pytest` and `tox` by exporting the variable:
```

Comment:
I'd add the relevant info in CONTRIBUTING, in the section where we talk about running tests.

Reply:
Updated. Let me know what you think.